lib/std/crypto/gimli.zig

```diff
@@ -104,15 +104,21 @@ pub const Hash = struct {
     state: State,
     buf_off: usize,
 
+    pub const block_length = State.RATE;
+
     const Self = @This();
 
-    pub fn init() Self {
-        return Self{
-            .state = State{
-                .data = [_]u32{0} ** (State.BLOCKBYTES / 4),
-            },
-            .buf_off = 0,
+    pub fn reset(self: *Self) void {
+        self.state = State{
+            .data = [_]u32{0} ** (State.BLOCKBYTES / 4),
         };
+        self.buf_off = 0;
+    }
+
+    pub fn init() Self {
+        var d: Self = undefined;
```

Comment:
I'm not sure if this pattern will be optimized.
A better choice might be to have `reset` do: `self.* = Hash.init()`

Reply:
This is the pattern used in the `md5`, `sha1`, `sha2` and `sha3` implementations, but I can certainly updated these as well.

Reply:
Done.